janusgraph-lucene/src/main/java/org/janusgraph/diskstorage/lucene/LuceneIndex.java

```diff
@@ -207,44 +207,42 @@ private Directory getStoreDirectory(String store) throws BackendException {
 
     private IndexWriter getWriter(String store, KeyInformation.IndexRetriever informations) throws BackendException {
         Preconditions.checkArgument(writerLock.isHeldByCurrentThread());
-        IndexWriter writer = writers.get(store);
-        if (writer == null) {
-            final LuceneCustomAnalyzer analyzer = delegatingAnalyzerFor(store, informations);
+        IndexWriter writer = writers.computeIfAbsent(store, key -> {
+            final LuceneCustomAnalyzer analyzer = delegatingAnalyzerFor(key, informations);
             final IndexWriterConfig iwc = new IndexWriterConfig(analyzer);
             iwc.setOpenMode(IndexWriterConfig.OpenMode.CREATE_OR_APPEND);
             try {
-                writer = new IndexWriter(getStoreDirectory(store), iwc);
-                writers.put(store, writer);
-            } catch (final IOException e) {
-                throw new PermanentBackendException("Could not create writer", e);
+                return new IndexWriter(getStoreDirectory(store), iwc);
+            } catch (final IOException | BackendException e) {
+                log.error("Could not create writer", e);
+                return null;
```

Comment:
I personally prefer the old style, i.e. without using `computeIfAbsent`. Now if the user has a higher log level, this root cause will not be available to users.